EventProc/src/HcalDigiProducer.cxx

```diff
@@ -24,21 +24,24 @@ namespace ldmx {
         NUM_SIDE_TB_HCAL_LAYERS_ = ps.getInteger("num_side_tb_hcal_layers");
         STRIPS_SIDE_LR_PER_LAYER_ = ps.getInteger("strips_side_lr_per_layer");
         NUM_SIDE_LR_HCAL_LAYERS_ = ps.getInteger("num_side_lr_hcal_layers");
+        SUPER_STRIP_SIZE_ = ps.getInteger("super_strip_size");
         readoutThreshold_ = ps.getInteger("readoutThreshold");
         meanNoise_   = ps.getDouble("meanNoise");
         mev_per_mip_ = ps.getDouble("mev_per_mip");
         pe_per_mip_  = ps.getDouble("pe_per_mip");
         doStrip_     = ps.getInteger("doStrip");
         noiseGenerator_ = new NoiseGenerator(meanNoise_,false);
-        noiseGenerator_->setNoiseThreshold(readoutThreshold_);
+        //noiseGenerator_->setNoiseThreshold(readoutThreshold_);
+        noiseGenerator_->setNoiseThreshold(1); // hard-code this number, create noise hits for non-zero PEs! 
     }
```

Comment:
Should we enforce that the number of NUM_BACK_HCAL_LAYERS_  is a multiple of SUPER_STRIP_SIZE_ , i.e. NUM_BACK_HCAL_LAYERS_  %SUPER_STRIP_SIZE_ = 0? It seems strange to have a fraction of a super strip, and that causes problems when we calculate the strip id.

Reply:
So to do what you are proposing is a little bit tricky -- which is why I went with this more flexible route.  

The core issue is that the information is spread out in different places and cannot necessarily be accessed at a given step.  The gdml file (which has an awkward size layer of 3.1 m now -> 62 strips) can only be accessed during the SIM step.  Hence we define a bar size in the HCal SD.
At RECO level, we don't know anything about the geometry anymore, so I was using the strip ID defined in the SIM step as a proxy for aggregating into superstrips.  If you want to do what you are suggesting, a parameter to the RECO job would have to be the transverse size of the HCal layers for a given geometry, which is a little backwards IMO.

The logic now is:
- gdml defines the geometry
- sim hit SD strip ID is defined at the SIM level using gdml geometry information
- reco hits and (super)strip ID aggregate sim hits using their strip ID

What you are proposing is:
- define the reco strip size and ID based on the geometry, but the issue is that at the RECO step, one does not have access to the geometry unless you hardcode it as a parameter

I'm not sure which is better.  In the former case, you can handle any geometry but can have partial superstrips while in the latter you have to enforce nice round numbers but is geometry dependent.